src/main/java/build/buildfarm/server/ByteStreamService.java

```diff
@@ -262,9 +262,11 @@ private void writeOperationStream(
         String operationStream = parseOperationStream(resourceName);
 
         OutputStream outputStream = instance.getStreamOutput(operationStream);
-        outputStream.write(request.getData().toByteArray());
-        if (request.getFinishWrite()) {
-          outputStream.close();
+        if (outputStream != null) {
```

Comment:
Right now I still see the null check on GitHub, so I'll wait with merging this PR.